corehq/motech/repeaters/expression/repeaters.py

```diff
@@ -0,0 +1,47 @@
+from django.utils.translation import ugettext_lazy as _
+
+from memoized import memoized
+
+from dimagi.ext.couchdbkit import DictProperty
+
+from corehq.apps.userreports.filters.factory import FilterFactory
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.motech.repeaters.expression.repeater_generators import (
+    ExpressionPayloadGenerator,
+)
+from corehq.motech.repeaters.models import Repeater
+from corehq.toggles import EXPRESSION_REPEATER
+
+
+class BaseExpressionRepeater(Repeater):
+    """Uses a UCR dict expression to send a generic json response
+    """
+    class Meta:
+        app_label = 'repeaters'
+
+    configured_filter = DictProperty()
+    configured_expression = DictProperty()
+    payload_generator_classes = (ExpressionPayloadGenerator,)
+
+    @classmethod
+    def available_for_domain(cls, domain):
+        return EXPRESSION_REPEATER.enabled(domain)
+
+    def allowed_to_forward(self, payload):
+        return FilterFactory.from_spec(self.configured_filter)(payload.to_json())
```

Comment:
I think it'd probably be worth `memoizing` the compiled filter. (assuming the repeater object itself has a lifecycle longer than 1 request... otherwise it doesn't matter)